### ISSUE TITLE
Chopper Generator build error "Expected to find ')'"

### ISSUE THREAD

**Issue by morzan1001** [lejard-h/chopper#249]
Hi guys, 

I tried to update my project from chopper 3 to chopper 4. I know of course that Chopper4 is still imm prerelease status, but due to other dependencies an upgrade was unfortunately necessary. Now I tried to create a new chopper script with the command `flutter packages pub run build_runner build --delete-conflicting-outputs` and errors occur. 

Specifically, I get as feedback: 

```
[SEVERE] chopper_generator:chopper_generator on lib/authentication/resources/services/rest_api_service.dart (cached):

Could not format because the source could not be parsed:

line 7, column 49 of .: Expected to find ')'.
  ╷
7 │ @override Future<Response<dynamic>> login(String* body) { final $url = 'api/v1/token';
  │                                                 ^
  ╵
line 38, column 69 of .: Expected to find ')'.
   ╷
38 │ @override Future<Response<dynamic>> getUserFeedbackForAnEvent(String* id) { final $url = 'api/v1/event/$id';
   │                                                                     ^
   ╵
line 35, column 63 of .: Expected to find ')'.
   ╷
35 │ @override Future<Response<dynamic>> getUserFeedbackById(String* id) { final $url = 'api/v1/feedback/$id';
   │                                                               ^
   ╵
[SEVERE] Failed after 400ms
```

The dart file we are talking about here looks like this: 

```
import 'package:chopper/chopper.dart';

part 'rest_api_service.chopper.dart';

@ChopperApi(baseUrl: 'api/v1')
abstract class RestApiService extends ChopperService {
  //Authenticate
  @Get(path: '/token')
  Future<Response> login(
    @Header('authorization') String body,
  );
  @Get(path: '/alarm')
  Future<Response> getAllAlarms();
  @Get(path: '/alarm/1')
  Future<Response> getSpecficAlarm();
  @Get(path: '/alarm/new')
  Future<Response> getNewAlarms();
  @Get(path: '/event')
  Future<Response> getAllEvents();
  @Get(path: '/termine/1')
  Future<Response> getSpecficEvent();
  @Get(path: '/termine/new')
  Future<Response> getNewEvent();
  @Get(path: '/user')
  Future<Response> getUserFunctionAndGroupsStatus();
  @Get(path: '/feedback/{id}')
  Future<Response> getUserFeedbackById(@Path() String id);
  @Get(path: '/event/{id}')
  Future<Response> getUserFeedbackForAnEvent(@Path() String id);
  @Get(path: '/groups')
  Future<Response> getAllGroups();
  @Get(path: '/functions')
  Future<Response> getAllFunctions();
  @Post(path: '/alarm/1')
  Future<Response> postUserFeedbackOnAlarm();
  @Post(path: '/termine/')
  Future<Response> postUserFeedbackOnAppointment();
  @Post(path: '/Status')
  Future<Response> postUserStatus();
  @Post(path: '/Groups')
  Future<Response> postUserFeedbackOnGroups();
  @Post(path: '/Functions')
  Future<Response> postUserFeedbackOnFunctions();

  static RestApiService create([ChopperClient client]) {
    return _$RestApiService(client);
  }
}
```

Maybe one of you has an idea what I am missing here, it would be cool if you could give me some feedback. 

Many greetings

**Comment by JEuler** [comment]
https://github.com/lejard-h/chopper/issues/238

Hi! You need to use 2.12.0 SDK

**Comment by morzan1001** [comment]
Thanks for the response. As you can see i have installed Dart 2.12.3
```
flutter upgrade
Flutter is already up to date on channel stable
Flutter 2.0.5 • channel stable • https://github.com/flutter/flutter.git
Framework • revision adc687823a (6 days ago) • 2021-04-16 09:40:20 -0700
Engine • revision b09f014e96
Tools • Dart 2.12.3
```
I also have no problems in `flutter doctor`:

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.0.5, on macOS 11.2.3 20D91 darwin-x64, locale
    de-DE)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.1)
[✓] VS Code (version 1.55.2)
[✓] Connected device (1 available)

• No issues found!
```

**Comment by JEuler** [comment]
You need to use it in your `pubspec.yaml` file for the project :) 

```
environment:
  sdk: '>=2.12.0 <3.0.0'
```

**Comment by morzan1001** [comment]
i have tried that but there are some problems. I use `build_runner` as a dev_dependency to build the chopper (and moor) files. To upgrade that package to Nullsafty the dependency `code_builder ^4.0.0` is needed. So far so good, but `chopper_generator` needs ` code_builder ^3.7.0`, an older version of code_builder and is there for not compatible with `build_runner >=2.0.0`. Do you have any suggestion what to use instead of `build_runner` to generate the chopper dart files?

**Comment by JEuler** [comment]
No, I have the following working setup:

```
dependency_overrides:
  analyzer: ^1.4.0
  hive_generator:
  build_runner: ^1.12.2
  quiver: ^3.0.1
```

```
chopper: ^4.0.0-nullsafety.1
chopper_generator: ^4.0.0-nullsafety.1
```

And 
```
environment:
  sdk: '>=2.12.0 <3.0.0'
```

We will work on the chopper_generator upgrade soon. #248

**Comment by morzan1001** [comment]
Thank you, then i will probably wait until  chopper_generator is released stable and uses the code_bulder ^4.0.0 because with an oder version my migration to Sound Nullsafty fails 😅.

**Comment by shashank090789** [comment]
@morzan1001  same issue I am facing after updating the chopper to 4.0.1.
flutter version is : 2.2.0
